src/ImageSharp/Common/Memory/BufferPointer{T}.cs

```diff
@@ -74,6 +74,20 @@ public BufferPointer(T[] array, void* pointerToArray)
         public IntPtr PointerAtOffset { get; private set; }
 
         /// <summary>
+        /// Gets the element at the specified position.
+        /// </summary>
+        /// <param name="index">The index from the start of this Pointer to the required element.</param>
+        /// <returns>The <see typeparam="T"/> at the specified position.</returns>
+        public T this[int index]
```

Comment:
This could potentially use ref returns + `Unsafe.Add<T>` @antonfirsov Am i correct?